src/app/server/Server.h

```diff
@@ -225,83 +233,7 @@ struct CommonCaseDeviceServerInitParams : public ServerInitParams
-        {
-            // WARNING: PersistentStorageOperationalKeystore::Finish() is never called. It's fine for
-            //          for examples and for now.
-            ReturnErrorOnFailure(sPersistentStorageOperationalKeystore.Init(this->persistentStorageDelegate));
-            this->operationalKeystore = &sPersistentStorageOperationalKeystore;
-        }
-
-        // OpCertStore can be injected but default to persistent storage default
-        // for simplicity of the examples.
-        if (this->opCertStore == nullptr)
-        {
-            // WARNING: PersistentStorageOpCertStore::Finish() is never called. It's fine for
-            //          for examples and for now, since all storage is immediate for that impl.
-            ReturnErrorOnFailure(sPersistentStorageOpCertStore.Init(this->persistentStorageDelegate));
-            this->opCertStore = &sPersistentStorageOpCertStore;
-        }
-
-        // Injection of report scheduler WILL lead to two schedulers being allocated. As recommended above, this should only be used
-        // for IN-TREE examples. If a default scheduler is desired, the basic ServerInitParams should be used by the application and
-        // CommonCaseDeviceServerInitParams should not be allocated.
-        if (this->reportScheduler == nullptr)
-        {
-            reportScheduler = &sReportScheduler;
-        }
-
-        // Session Keystore injection
-        this->sessionKeystore = &sSessionKeystore;
-
-        // Group Data provider injection
-        sGroupDataProvider.SetStorageDelegate(this->persistentStorageDelegate);
-        sGroupDataProvider.SetSessionKeystore(this->sessionKeystore);
-        ReturnErrorOnFailure(sGroupDataProvider.Init());
-        this->groupDataProvider = &sGroupDataProvider;
-
-#if CHIP_CONFIG_ENABLE_SESSION_RESUMPTION
-        ReturnErrorOnFailure(sSessionResumptionStorage.Init(this->persistentStorageDelegate));
-        this->sessionResumptionStorage = &sSessionResumptionStorage;
-#else
-        this->sessionResumptionStorage = nullptr;
-#endif
-
-        // Inject access control delegate
-        this->accessDelegate = Access::Examples::GetAccessControlDelegate();
-
-        // Inject ACL storage. (Don't initialize it.)
-        this->aclStorage = &sAclStorage;
-
-#if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
-        ChipLogProgress(AppServer, "Initializing subscription resumption storage...");
-        ReturnErrorOnFailure(sSubscriptionResumptionStorage.Init(this->persistentStorageDelegate));
-        this->subscriptionResumptionStorage = &sSubscriptionResumptionStorage;
-#else
-        ChipLogProgress(AppServer, "Subscription persistence not supported");
-#endif
-
-#if CHIP_CONFIG_ENABLE_ICD_CIP
-        if (this->icdCheckInBackOffStrategy == nullptr)
-        {
-            this->icdCheckInBackOffStrategy = &sDefaultICDCheckInBackOffStrategy;
-        }
-#endif
-
-        return CHIP_NO_ERROR;
-    }
+    CHIP_ERROR InitializeStaticResourcesBeforeServerInit();
```

Comment:
De-inlining hides the fact that this is a big hack not to be expanded.

See the big warning above on line 214:

```
 * WARNING: DO NOT replicate the pattern shown here of having a subclass of ServerInitParams
 *          own the resources outside of examples. This was done to reduce the amount of change
 *          to existing examples while still supporting non-example versions of the
 *          resources to be injected.
 * ```